+nosnoc/+dcs/stewart.m

```diff
@@ -0,0 +1,118 @@
+classdef stewart < nosnoc.dcs.base
+    properties
+        model
+
+        theta
+        theta_sys
+        lambda
+        lambda_sys
+        mu
+        mu_sys
+
+        z_all
+
+        f_x
+        g_Stewart
+
+        sys_idx
+
+        dims
+
+        % TODO some of these should be in base perhaps
+        f_x_fun
+        f_q_fun
+        g_z_fun
+        g_alg_fun
+        g_switching_fun
+        g_Stewart_fun
+        lambda00_fun
+        g_path_fun
+        g_comp_path_fun
+        g_terminal_fun
+        f_q_T_fun
+        f_lsq_x_fun
+        f_lsq_u_fun
+        f_lsq_T_fun
+    end
+
+    methods
+        function obj = stewart(model)
+            obj.model = model;
+            obj.dims = model.dims;
+        end
+        
+        function generate_variables(obj, opts)
+            import casadi.*
+            dims = obj.dims;
+            % dimensions
+            dims.n_theta = sum(obj.dims.n_f_sys); % number of modes
+            dims.n_lambda = dims.n_theta;
+            dims.n_mu = dims.n_sys;
+            idx = 1;
+            for ii = 1:dims.n_sys
+                sys_idx{ii} = idx:(idx + obj.dims.n_f_sys(ii)-1);
+                idx = idx + obj.dims.n_f_sys(ii);
+                ii_str = num2str(ii);
+                % define theta (Filippov multiplers)
+                obj.theta_sys{ii} = define_casadi_symbolic(opts.casadi_symbolic_mode,['theta_' ii_str],obj.dims.n_f_sys(ii));
+                obj.theta = [obj.theta;obj.theta_sys{ii}];
+                % define mu_i (Lagrange multipler of e'theta =1;)
+                obj.mu_sys{ii} = define_casadi_symbolic(opts.casadi_symbolic_mode,['mu_' ii_str],1);
+                obj.mu = [obj.mu;obj.mu_sys{ii}];
+                % define lambda_i (Lagrange multipler of theta >= 0;)
+                obj.lambda_sys{ii} = define_casadi_symbolic(opts.casadi_symbolic_mode,['lambda_' ii_str],obj.dims.n_f_sys(ii));
+                obj.lambda = [obj.lambda;obj.lambda_sys{ii}];
+            end
+
+            % symbolic variables z = [theta;lambda;mu_Stewart];
+            obj.z_all = [obj.theta;obj.lambda;obj.mu;obj.model.z];
+            obj.dims = dims;
+        end
+
+        function generate_equations(obj, opts)
+            import casadi.*
+            model = obj.model;
+            dims = obj.dims;
+            
+            obj.f_x = zeros(dims.n_x,1);
+            for ii = 1:dims.n_sys
+                obj.f_x = obj.f_x + model.F{ii}*obj.theta_sys{ii};
+                obj.g_Stewart{ii} = -model.S{ii}*model.c{ii};
```

Comment:
what if g_Stewart given, and model.S and model.c are not, e.g. in  Voronoi type PSS?

Reply:
Yep, this is fixed.